docs/ocis/adr/0027-new-share-jail.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "27. New Share Jail"
+date: 2024-02-21T15:19:00+01:00
+weight: 27
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/ocis/adr
+geekdocFilePath: 0027-new-share-jail.md
+---
+
+* Status: draft
+* Deciders: [@butonic](https://github.com/butonic), [@rhafer](https://github.com/rhafer), [@dragotin](https://github.com/dragotin)
+* Date: 2024-02-21
+
+## Context and Problem Statement
+
+The oCIS share jail is a space that contains all accepted / synced shares of a user. In contrast to a personal or project space that contains actual resources, the share jail space only contains references pointing to shared resources. The root directory only consists of mountpoints that actually represent resources in other spaces. On the WebDAV API clients expect an `oc:fileid` property to identify resources in other API endpoints, eg. the libregraph `/me/sharedWithMe` endpoint. 
+
+Currently, we construct the `oc:fileid` from the pattern `{shareproviderid}${sharespaceid}!{sharemountid}`. `{shareproviderid}`and `{sharespaceid}` are both hardcoded to `a0ca6a90-a365-4782-871e-d44447bbc668`. The `{sharemountid}` itself uses the pattern `{shared-resource-providerid}:{shared-resource-spaceid}:{shareid}`.
+
+Since a resource can be shared to the same user in multiple ways (a group share and a user share) we deduplicate the two shares and only show one mountpoint in the share jail root. This is where this solution starts to fall apart:
+* When accepting, mounting or syncing a share we implicitly have to accept all shares
+* Each share has a different `{shareid}`, so we currently look up the oldest share and use it to build the `oc:fileid`
+* Consequently, when the oldest share is revoked the `oc:fileid` changes.
+
+We need to build the `oc:fileid` from a more stable pattern.
+
+### Shareid
+
+The WebDAV PROPFIND response also contains a `oc:shareid` which currentld
```

Comment:
This sentence is incomplete. I guess you wanted to write something about `oc:shareid` not being set consistently.